### PR TITLE
Fixes for cc-shim

### PR DESCRIPTION
**Pull request by amshinde**

### CODE REVIEW

File: shim/utils.c [hunk 1]
```diff
@@ -118,9 +118,10 @@ set_big_endian_32(uint8_t *buf, uint32_t val)
  *
  * \return Unsigned 32 bit network ordered integer
  */
-uint32_t 
-get_big_endian_32(char *buf) {
-	return (uint32_t)(buf[0] >> 24 | buf[1] >> 16 | buf[2] >> 8 | buf[3]);
+uint32_t
+get_big_endian_32(uint8_t *buf)
```
Comment (jodh-intel) [inline shim/utils.c]: `const`?

File: shim/utils.c [hunk 2]
```diff
@@ -144,7 +145,7 @@ set_big_endian_64(uint8_t *buf, uint64_t val)
  * \return Unsigned 64 bit network ordered integer
  */
 uint64_t
-get_big_endian_64(char *buf) {
+get_big_endian_64(uint8_t *buf) {
```
Comment (jodh-intel) [inline shim/utils.c]: `const`?

File: shim/shim.c [hunk 3]
```diff
@@ -368,7 +368,7 @@ read_IO_message(struct cc_shim *shim, uint64_t *seq, ssize_t *stream_len) {
 			}
 		}
 	}
-	*seq = get_big_endian_64(buf);
+	*seq = get_big_endian_64((uint8_t*)buf);
```
Comment (dlespiau) [inline shim/shim.c]: I think it'd make sense to declare buf as uint8_t * instead of having to cast it?

Comment (amshinde) [inline shim/shim.c]: I have used string functions at some other places. I would have to cast to char* there.

Comment (dlespiau) [inline shim/shim.c]: Hum, I don't see any string function on buf in read_IO_message()?

Comment (dlespiau) [inline shim/shim.c]: but then the return value is used elsewhere in a atoi(buf). I'd argue the cast is probably better there (interpreting part of the byte stream as something else), but it doesn't really matter, let's drop this.

**Comment by dlespiau** [comment]
lgtm

**Comment by grahamwhaley** [comment]
lgtm
presuming @jodh-intel is happy with rework

**Comment by chavafg** [comment]
qa-passed

**Pull request merged**